### PR TITLE
PullRequst for Issue6: Server App crashed when sending request from TCPClient

### PR DESCRIPTION
**Pull request by XuetaoLiu-S**
https://github.com/acquaman/AcquamanDataServer/issues/6

The crash is caused by the empty count field sent from the TCPClient.

Solution:
- Added a protection in the server app, if no data field, the request will be ignored
- for StartTime + Count request type, if count is not provided, default it as 1.

---

Note: 
This pull request is based on Branch Issue38, which should be reviewed first.

### CODE REVIEW

**Comment by iainworkman** [comment]
Looks good to me. I would maybe suggest an addition to the documentation for start time + count request type which indicates that it the count value floors at 1, but other than that everything looks great.

File: source/AMDSThreadedTCPDataServer.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+	thread_ = new QThread;
+	server_ = new AMDSTCPDataServer;
+
+	connect(thread_, SIGNAL(started()), this, SLOT(onThreadStarted()));
+	connect(this, SIGNAL(startServer(QString,quint16)), server_, SLOT(start(QString,quint16)));
+	connect(server_, SIGNAL(error(quint8,quint16,QString)), this, SIGNAL(error(quint8,quint16,QString)));
+
+	server_->moveToThread(thread_);
+	thread_->start(QThread::LowPriority);
```
Comment (dretrex) [inline source/AMDSThreadedTCPDataServer.cpp:17]: Can you explain the idea behind the priority choice?  Is there a particular reason for a low priority vs any of the other ones?

File: source/AMDSThreadedTCPDataServer.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+AMDSThreadedTCPDataServer::~AMDSThreadedTCPDataServer()
+{
+	if (thread_->isRunning())
+		thread_->terminate();
```
Comment (dretrex) [inline source/AMDSThreadedTCPDataServer.cpp]: @iainworkman noticed this in a diffferent pull request, but you should probably be using `QThread::quit()` instead.

**Comment by dretrex** [comment]
Rather than say that the default value is 1, it's more appropriate to say that the value must be _at least_ 1 because if it's 0 then it ceases to work correctly.

**Comment by dretrex** [comment]
This can be merged.  @XuetaoLiu-CLS, I'll let you do the final merge/delete branch step on these pull requests since you know the appropriate ordering.  @iainworkman and I will make our comments and give them the thumbs up and then you can merge in at your convenience.  Sound good?

**Comment by dretrex** [comment]
Time Estimate: 1 minute

**Pull request merged**